crates/chia-consensus/src/merkle_set.rs

```diff
@@ -147,567 +185,892 @@ fn radix_sort(range: &mut [[u8; 32]], depth: u8) -> ([u8; 32], NodeType) {
+            tree.hash_cache[tree.hash_cache.len() - 1]
+        );
+    }
+
+    #[test]
+    fn test_merkle_right_edge() {
+        let a = [
+            0x40, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0,
+        ];
+        let b = [
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0xff, 0xff, 0xff, 0xff,
+        ];
+        let c = [
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0xff, 0xff, 0xff, 0xfe,
+        ];
+        let d = [
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0xff, 0xff, 0xff, 0xfd,
+        ];
+
+        let mut expected = hashdown(&[1, 1], &c, &b);
+        expected = hashdown(&[1, 2], &d, &expected);
+
+        for _i in 0..253 {
+            expected = hashdown(&[0, 2], &BLANK, &expected);
+        }
+
+        expected = hashdown(&[1, 2], &a, &expected);
+
+        assert_eq!(compute_merkle_set_root(&mut [a, b, c, d]), expected);
+        let (root, tree) = generate_merkle_tree(&mut [a, b, c, d]);
+        assert_eq!(root, expected);
+        assert_eq!(root, compute_merkle_set_root(&mut [a, b, c, d]));
+        assert_eq!(tree.leaf_vec.len(), 4);
+        // this tree looks like this:
+        //           o
+        //          / \
+        //         a   o
+        //            / \
+        //           E   o
+        //              / \
+        //             E   o
+        //                 .
+        //                 .
+        //                 .
+        //                 o
+        //                / \
+        //               d   o
+        //                  / \
+        //                 c   b
+
+        let ArrayTypes::Middle { children } = tree.nodes_vec.last().unwrap() else {
+            panic!("expected middle node");
+        };
+        let ArrayTypes::Leaf { .. } = tree.nodes_vec[children.0] else {
+            panic!("expected leaf");
+        };
+        // generate proof of inclusion for every node
+        let (included, _proof) = tree.generate_proof(d).unwrap();
+        assert!(included);
+        let (included, _proof) = tree.generate_proof(a).unwrap();
+        assert!(included);
+        let (included, proof) = tree.generate_proof(c).unwrap();
+        assert!(included);
+        assert_eq!(tree.hash_cache.len(), tree.nodes_vec.len());
+        let rebuilt = deserialize_proof(&proof).unwrap();
+        assert_eq!(
+            rebuilt.hash_cache[rebuilt.hash_cache.len() - 1],
+            tree.hash_cache[tree.hash_cache.len() - 1]
+        );
```

Comment:
this is just checking the root hash, right? I would seem reasonable to have at least one test to check the uncle hashes as well

Reply:
I can add these tests later